tcms/static/js/testrun_actions.js

```diff
@@ -259,91 +269,61 @@ function updateRunStatus(object_pk, value, callback) {
   });
 }
 
-var updateCaseRunStatus = function(e) {
-  e.stopPropagation();
-  e.preventDefault();
-  var container = jQ(this).parents().eq(3);
-  var parent = container.parent();
-  var title = parent.prev();
-  var link = title.find('.expandable')[0];
-  var parameters = Nitrate.Utils.formSerialize(this);
-  var object_pk = parameters['object_pk'];
-  var value = parameters['value'];
-
-  // Callback when
-  var callback = function(t) {
-    // Update the contents
-    const value = parameters['value']
-    if (value != '') {
-      // Update the case run status icon
-      var crs = Nitrate.TestRuns.CaseRunStatus;
-      title.find('.execution_status_icon').each(function(index) {
-        for (i in crs) {
-          jQ(this).removeClass(crs[i].icon);
-        }
-        const execution = Nitrate.TestRuns.CaseRunStatus.find(tcs => tcs.pk === value);
-        jQ(this).addClass(execution.icon);
-        jQ(this).css('color', execution.color)
-      });
+function updateExecutionStatus(executionId, caseId, executionStatusPk) {
 
-      // Update related people
-      var usr = Nitrate.User;
-      title.find('.link_tested_by').each(function(i) {
-        jQ(this).html(usr.username);
-      });
-    }
+  const container = jQ('.update_form').parents().eq(3)
+  const parent = container.parent()
+  const title = parent.prev()
+  const link = title.find('.expandable')[0];
+
+  const comment = jQ('td > textarea[name=comment]').val()
+
+  setContainerToLoading(container, caseId)
+
+  if (comment) {
+    jsonRPC('TestExecution.add_comment', [executionId, comment], () => updateCommentsCount(caseId, true));
+  }
+
+  // Update the object when changing the status
+  jsonRPC('TestExecution.update', [executionId, {'status': executionStatusPk}], () => {
+    // Update the case run status icon
+    // TODO: remove this global variable in favour of API call
+    var executionStatuses = Nitrate.TestRuns.CaseRunStatus;
+    title.find('.execution_status_icon').each((_, icon) => {
+      executionStatuses.forEach(status => jQ(icon).removeClass(status.icon))
+
+      const execution = executionStatuses.find(tcs => tcs.pk === executionStatusPk)
+      jQ(icon).addClass(execution.icon)
+      jQ(icon).css('color', execution.color)
+    });
 
+    // Update related people
+    title.find('.link_tested_by').each(() => jQ(this).html(Nitrate.User.username));
     // Mark the case run to mine
     if (!title.is('.mine')) {
       title.addClass('mine');
     }
 
     // Blind down next case
     fireEvent(link, 'click');
-    if (jQ('#id_check_box_auto_blinddown').attr('checked') && parameters['value'] != '') {
+    if (jQ('#id_check_box_auto_blinddown').attr('checked') && executionStatusPk) {
       var next_title = parent.next();
       if (!next_title.length) {
         return false;
       }
       if (next_title.next().is(':hidden')) {
         fireEvent(next_title.find('.expandable')[0], 'click');
       }
-    } else {
-      fireEvent(link, 'click');
-    }
-  };
```

Comment:
I'm not exactly sure why the else block was removed but the current result is that the existing TE is collapsed and the next one isn't opened automatically if I haven't checked the "Auto" checkbox above the grid.